### DISCUSSION TITLE
workflow_dispatch: Passing Secrets in at runtime

### DISCUSSION THREAD

**Discussion by MikeRochfordTR** [community/community#45644] [Actions]
### Select Topic Area

Product Feedback

### Body

The workflow I am working on takes git credentials(user and pat) as input, as the person running the workflow doesn't own the repository. The workflow is kicked off via an internal site using credentials managed by my team. 

The issue we are running into is how to secure the passed git credentials from every showing up in the logs.  I have been banging my head against the wall for several days and can't find a solution. 

![image](https://user-images.githubusercontent.com/52175743/215285563-b23de8ad-8409-4f34-b094-2a4f2694b99d.png)

resulting log

![image](https://user-images.githubusercontent.com/52175743/215285592-e8b439eb-7fdd-4ca3-a452-6d1d38fcab82.png)

In this instance, both inputs had the value of 'unset' passed.

**Comment by ooaklee** [comment]
Hey @MikeRochfordTR, I hope you're doing well.

I wanted to check in with you to see if you're still experiencing the issue. I've developed an Action called [**Interactive Inputs**](https://github.com/boasiHQ/interactive-inputs/) that allows users to add runtime inputs to their workflows/actions and more. After reviewing your topic, I thought about adding an `isSecret` property to the `text` and `textarea` field types of the action. This feature would ensure that any values passed are appropriately masked when set. Would something like this have solved your use case?

![InteractiveInputs_LiveFileInputTypeSupport](https://github.com/user-attachments/assets/5f2a9c8b-fc3f-4bf5-a3e3-cbefddc1ad4b)
